### PR TITLE
CSHARP-4557: Add support for ContainsKey in LINQ3.

### PR DESCRIPTION
**Pull request by rstam**

### CODE REVIEW

File: src/MongoDB.Driver/Linq/Linq3Implementation/Translators/ExpressionToAggregationExpressionTranslators/MethodTranslators/ContainsKeyMethodToAggregationExpressionTranslator.cs [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+
+                AstExpression ast;
+                switch (dictionaryRepresentation)
+                {
+                    case DictionaryRepresentation.Document:
+                        if (keyExpression.Type != typeof(string))
+                        {
+                            throw new ExpressionNotSupportedException(expression, because: "ContainsKey requires key to be of type string when DictionaryRepresentation is: Document");
+                        }
+                        ast = AstExpression.Ne(AstExpression.Type(AstExpression.GetField(dictionaryTranslation.Ast, keyTranslation.Ast)), "missing");
```
Comment (rstam) [inline src/MongoDB.Driver/Linq/Linq3Implementation/Translators/ExpressionToAggregationExpressionTranslators/MethodTranslators/ContainsKeyMethodToAggregationExpressionTranslator.cs:52]: This expression was discussed on Slack. You can help me confirm that this is a correct way to test for the existence of a field in a document in the aggregation expression language.

File: tests/MongoDB.Driver.Tests/Linq/Linq3ImplementationTests/Jira/CSharp4557Tests.cs [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+            var collection = CreateCollection(linqProvider);
+
+            var queryable = collection
+                .AsQueryable()
+                .Select(x => x.Foo.ContainsKey("bar"));
+
+            if (linqProvider == LinqProvider.V2)
+            {
+                var exception = Record.Exception(() => Translate(collection, queryable));
+                exception.Should().BeOfType<NotSupportedException>();
```
Comment (rstam) [inline tests/MongoDB.Driver.Tests/Linq/Linq3ImplementationTests/Jira/CSharp4557Tests.cs:60]: Note that LINQ2 only supported ContainsKey in the filter language, not in the aggregation expression language.

File: src/MongoDB.Driver/Linq/Linq3Implementation/Translators/ExpressionToAggregationExpressionTranslators/MethodTranslators/ContainsKeyMethodToAggregationExpressionTranslator.cs (hunk 1 above)
Comment (JamesKovacs) [reply src/MongoDB.Driver/Linq/Linq3Implementation/Translators/ExpressionToAggregationExpressionTranslators/MethodTranslators/ContainsKeyMethodToAggregationExpressionTranslator.cs:52]: I tried it out in `mongosh` myself and that MQL behaves as expected.

**Review by JamesKovacs — APPROVED** [review]
LGTM

**Pull request closed without merging**